kuma/settings/common.py

```diff
@@ -1905,3 +1905,9 @@ def get_user_url(user):
 ADDITIONAL_NEXT_URL_ALLOWED_HOSTS = config(
     "ADDITIONAL_NEXT_URL_ALLOWED_HOSTS", default=None
 )
+
+# The PROXY_DEPTH is used to avoid client IP spoofing by extracting a trustworthy
+# client IP from the X-Forwarded-For header. For stage and production, there are
+# two proxies, the CDN and the public-facing ELB. If set to zero, the first IP
+# in the X-Forwarded-For header is selected as the client IP.
+PROXY_DEPTH = config("PROXY_DEPTH", cast=int, default=0)
```

Comment:
I made the default `0` because that's the way it works currently (it always selects the first IP in the list), but that's probably a bad decision. For stage/production it should be `2`, and I suppose for local development it should be just 0?

Reply:
Make it 2 by default, so that the defaults are geared for secure production. And in development it won't matter because of the `len` check. In `docker-compose`, the `len(forwarded_ips)` will be 1. 
Which brings us to the conclusion that it should be something like this maybe:
```python
if len(forwarded_for_ips) > 1:
    client_ip = forwarded_for_ips[-1 * settings.PROXY_DEPTH]
else:
    # E.g. local development where you don't have an ELB and or CDN.
    client_ip = forwarded_for_ips[0]
request.META["REMOTE_ADDR"] = str(ip_address(client_ip))
```

Reply:
In general, I definitely think the defaults should be for the most secure thing which is what you want prod to be. 
And you should use explicit exceptions to make it less secure. Thankfully we have the power of that list of env settings in `docker-compose` so we can have the whole of kuma default nicely for local dev but also be secure in prod by default. 

In other words; you shouldn't need to remember to set some env var in mdn/infra to make it secure. Typos and confusion can happy so easily. 
Which is what we do with `DEBUG` by the way.

Reply:
@peterbe Agreed, using the production settings by default (so 2 in our case) is much better. For local development, the  `X-Forwarded-For` header isn't present in the requests by default, but I'll re-factor my code to handle the cases better.